sql/core/src/main/scala/org/apache/spark/sql/execution/command/resources.scala

```diff
@@ -37,7 +37,7 @@ case class AddJarCommand(path: String) extends RunnableCommand {
   }
 
   override def run(sparkSession: SparkSession): Seq[Row] = {
-    sparkSession.sessionState.addJar(path)
+    sparkSession.sharedState.addJar(path)
```

Comment:
we should also update the document of `AddJarCommand`, and mention that it's cross-session